src/com/lms/model/Loan.java

```diff
@@ -60,50 +49,21 @@ public void setDueDate(LocalDate dueDate) {
 
 	@Override
 	public int hashCode() {
-		final int prime = 31;
-		int result = 1;
-		result = prime * result + ((book == null) ? 0 : book.hashCode());
-		result = prime * result + ((borrower == null) ? 0 : borrower.hashCode());
-		result = prime * result + ((branch == null) ? 0 : branch.hashCode());
-		result = prime * result + ((dateOut == null) ? 0 : dateOut.hashCode());
-		result = prime * result + ((dueDate == null) ? 0 : dueDate.hashCode());
-		return result;
+		return Objects.hash(book, borrower, branch);
 	}
 
 	@Override
-	public boolean equals(Object obj) {
-		if (this == obj)
+	public boolean equals(final Object obj) {
```

Comment:
Why do you do a final Object obj?

Reply:
As I've mentioned, I have Eclipse set to add `final` modifiers to every variable that is set to at most once, including method parameters, whenever I save. I excluded most of those changes from the commit (as at-this-point unnecessary churn), but here I would have had to edit the hunk to keep it out.